### ISSUE TITLE
Gradient Descent Tests

### ISSUE THREAD

**Issue by zedfly** [felix642/1chipML#16]
1. Define function with small number of variable and compute the min
2. On the edge ML model training (e.g., neural network on a relatively small problem)
     Implique le backpropagation, pas forcement reseaux de neuronnes
     Model avec hyperparametre pour voir ce qui donne les bonne prediction

3. Curve fitting from measurements (e.g. Simple short-term predictions/estimations)
     Recoit mesure sur microcontroleur (ex: Temperature)
     Curve fitting : donner valeur entre mesuration, ou apres mesuration

4 Convex optimization (e.g., for industry process optimization) 
     Trouver fct convex avec sens industriel
     Pas trop de param sinon compliquer
